builder/llpcBuilderImpl.cpp

```diff
@@ -321,6 +321,47 @@ Value* BuilderImplBase::Scalarize(
     return pResult;
 }
 
+// =====================================================================================================================
+// Helper method to scalarize in pairs a possibly vector unary operation. The callback function is called
+// with vec2 input, even if the input here is scalar.
+Value* BuilderImplBase::ScalarizeInPairs(
+    Value*                        pInValue,   // [in] Input value
+    std::function<Value*(Value*)> callback)   // [in] Callback function
+{
+    if (auto pVecTy = dyn_cast<VectorType>(pInValue->getType()))
+    {
+        Value* pInElements = CreateShuffleVector(pInValue, pInValue, { 0, 1 });
```

Comment:
pInElements -> pComps

Reply:
Why? "Element" is the LLVM term for what a vector is made of.

Reply:
I think element in LLVM is unclear. Vector has elements and array has elements. In SPIR-V, those are treated as components (<n x comp>) and elements ([n x comp]). When you use component, we know this is for a vector operation. When use element, we denote this is for a array.

Reply:
I disagree with using GL/SPIR-V terminology here in the middle-end where it disagrees with LLVM terminology, especially as we are calling "CreateInsertElement" in this code. But I'll do it anyway so I can get this change in.

Reply:
Thanks for your understanding. Currently, LLPC serves SPIR-V, so it is better we can align the terminologies. Unlike LLVM, which has so many clients, LLPC's use is relatively limited. This is why Qun and I want to use SPIR-V naming in builder interface methods.